Realm/Realm/DatabaseTypes/RealmValue.cs

```diff
@@ -30,9 +30,10 @@
 namespace Realms
 {
     /// <summary>
-    /// A type that can represent any valid Realm data type. It is a valid type in and of itself,
-    /// which means that it can be used to declare a property of type <see cref="RealmValue"/> that
-    /// can hold any type.
+    /// A type that can represent any valid Realm data type (except for <see cref="IEmbeddedObject"/> and
+    /// <see cref="IAsymmetricObject"/> when being added to a realm). It is a valid type in and of itself, which
+    /// means that it can be used to declare a property of type <see cref="RealmValue"/> that can hold any type
+    /// (excluding the aforementioned exceptions).
```

Comment:
```suggestion
    /// A type that can represent any valid Realm data type. It is a valid type in and of itself, which
    /// means that it can be used to declare a property of type <see cref="RealmValue"/>
    /// except for <see cref="IEmbeddedObject"/> and <see cref="IAsymmetricObject"/>.
    /// If any of the latter two types is used as properties in an <see cref="IRealmObjectBase"/>
    /// inheritor, an exception is thrown when adding the such object to a realm.
```
You could give more info on the type of the exception raised.
You may also find better wording, but it seems simpler to me dropping that
> (except ... to a realm).

Reply:
Good update